digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/strategy/DefaultSectionCalculatorStrategy.scala

```diff
@@ -234,7 +234,7 @@ class DefaultSectionCalculatorStrategy extends RoadAddressSectionCalculatorStrat
    * @param pls Left or right side ProjectLinks with combined to check for continuity of addresses.
    */
   def validateAddresses(pls: Seq[ProjectLink]): Unit = {
-    if (pls.size > 1) {
+    if (pls.size > 1 && pls.head.originalStartAddrMValue == 0) {
       val it = pls.sliding(2)
       while (it.hasNext) {
         it.next() match {
```

Comment:
Tuon rivin 244 if lauseen voisi muotoilla uudestaan &&- operaattorilla

Reply:
nykyinen toteutus ohittaa pituudentarkistuksen, jos linkki on uusi, && -operaattori rikkoisi tämän toiminnallisuuden

Reply:
Scalassa tuo && operaattori taitaa olla “short-circuit evaluation" jolloin jos ensimmäinen ehto ei mene läpi niin toista ei edes testata. 

Mutta jos se halutaan pitää tuollaisena niin sen voisi korjata edes olemaan näin jolloin siitä saa helpommin selvää

```
if(1+1 == 2) {
    if(2+2 == 4) {
        doSomething()
    }
}
```